packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/ios/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,527 @@
+    currentRemediation!.proceed()
+}
+```
+
+### 5: Display the shared secret, QR Code, and request the code
+
+After selecting the authenticator, the server sends an enrollment request in a remediation of type `.enrollAuthenticator`. Enrolling uses a secret key shared between Okta and Google Authenticator that's used to generate and verify the generated time-based one-time passcodes (TOTP).
+
+The remediation includes the shared secret as both a QR Code and as text. It also includes a request to enter a Google Authenticator code after enrollment.
+
+The sign-on controller code for `handleEnrollAuthenticator(remdiation:)` works for many different authenticators because the `form` property of the remediation contains the information required to build the UI:
+
+```swift
+func handleEnrollAuthenticator(_ remdiation: Remediation) {
+    // Tell the caller to present a UI for enrolling in Google Authenticator.
+
+}
+```
+
+The authenticators related to the current request are in the `authenticators` property of the remediation. The `Authenticator` class represents an authenticator and includes a a `key` property that identifies it, and a display name for the UI in the `displayName` property.
+
+Associated information such as the QR Code and shared secret are accessed using capabilities.
+
+
+This utility function finds the Google Authenticator (`key` set to `"google_otp"`), and then loads an image for the QR Code and the shared secret:
+
+
+```swift
+// The information for constructing a UI for enrollment.
+typealias AuthenticatorEnrollment = (name: String?,
+                                     qrCode: UIImage?,
+                                     sharedSecret: String?)
+
+func authenticatorEnrollment() -> AuthenticatorEnrollment? {
+        guard let remediation = currentRemediation,
+              let authenticator = remediation.authenticators.filter({ $0.key == "google_otp"}).first,
+              let capability = authenticator.capabilities.filter({$0 is Capability.OTP}).first as? Capability.OTP,
+              let qrCode = capability.image,
+              let sharedSecret = capability.sharedSecret else {
+                  return nil
+              }
+
+        return (authenticator.displayName, qrCode, sharedSecret)
+    }
+```
+
+The model and view information for the one-time code entry field are nested in the forms of the remediation. First find a field with the `name` property set to `"credentials"` in the `form` property of the remediation. Then in the `form` property of that field, find the field with the `name` set to `"passcode"`:
+
+```swift
+typealias AuthenticatorCode = (codeField: Remediation.Form.Field,
+                               entryField: Remediation.Form.Field)
+
+    func authenticatorCodeField() -> AuthenticatorCode? {
+        guard let remediation = currentRemediation,
+              let codeField = remediation.form.fields.filter({ $0.name == "credentials"}).first,
```

Comment:
You can use `remediation["credentials"]` instead, since keyed subscripting is shorter.  Or, the dot-notation `"credentials.passcode"`.

Reply:
Code updated.